controllers/k8ssandra/medusa_reconciler_test.go

```diff
@@ -0,0 +1,235 @@
+
+	t.Log("check that dc2 was created")
+	require.Eventually(f.DatacenterExists(ctx, dc2Key), timeout, interval)
+
+	t.Log("check that remote seeds are set on dc2")
+	dc2 = &cassdcapi.CassandraDatacenter{}
+	err = f.Get(ctx, dc2Key, dc2)
+	require.NoError(err, "failed to get dc2")
+
+	t.Log("update dc2 status to ready")
+	err = f.PatchDatacenterStatus(ctx, dc2Key, func(dc *cassdcapi.CassandraDatacenter) {
+		dc.Status.CassandraOperatorProgress = cassdcapi.ProgressReady
+		dc.SetCondition(cassdcapi.DatacenterCondition{
+			Type:               cassdcapi.DatacenterReady,
+			Status:             corev1.ConditionTrue,
+			LastTransitionTime: metav1.Now(),
+		})
+	})
+	require.NoError(err, "failed to update dc2 status to ready")
+	checkMedusaObjectsCompliance(t, f, dc2, kc)
+
+	t.Log("check that the K8ssandraCluster status is updated")
+	require.Eventually(func() bool {
+		kc := &api.K8ssandraCluster{}
+		err = f.Get(ctx, kcKey, kc)
+		if err != nil {
+			t.Logf("failed to get K8ssandraCluster: %v", err)
+			return false
+		}
+
+		if len(kc.Status.Datacenters) != 2 {
+			return false
+		}
+
+		k8ssandraStatus, found := kc.Status.Datacenters[dc1Key.Name]
+		if !found {
+			t.Logf("status for datacenter %s not found", dc1Key)
+			return false
+		}
+
+		condition := findDatacenterCondition(k8ssandraStatus.Cassandra, cassdcapi.DatacenterReady)
+		if condition == nil || condition.Status == corev1.ConditionFalse {
+			t.Logf("k8ssandracluster status check failed: cassandra in %s is not ready", dc1Key.Name)
+			return false
+		}
+
+		if k8ssandraStatus.Reaper == nil || !k8ssandraStatus.Reaper.IsReady() {
+			t.Logf("k8ssandracluster status check failed: reaper in %s is not ready", dc1Key.Name)
+		}
+
+		k8ssandraStatus, found = kc.Status.Datacenters[dc2Key.Name]
+		if !found {
+			t.Logf("status for datacenter %s not found", dc2Key)
+			return false
+		}
+
+		condition = findDatacenterCondition(k8ssandraStatus.Cassandra, cassdcapi.DatacenterReady)
+		if condition == nil || condition.Status == corev1.ConditionFalse {
+			t.Logf("k8ssandracluster status check failed: cassandra in %s is not ready", dc2Key.Name)
+			return false
+		}
+
+		return true
+	}, timeout, interval, "timed out waiting for K8ssandraCluster status update")
+}
+
+// Check that all the Medusa related objects have been created and are in the expected state.
+func checkMedusaObjectsCompliance(t *testing.T, f *framework.Framework, dc *cassdcapi.CassandraDatacenter, kc *api.K8ssandraCluster) {
+	require := require.New(t)
+
+	// Check containers presence
+	require.True(f.ContainerExists(dc, "medusa-restore", true), "dc1 doesn't have medusa-restore init container")
+	// require.True(f.ContainerExists(dc, "medusa", false), "dc1 doesn't have medusa main container")
+
+	// Check containers Image
+	require.True(f.GetContainer(dc, "medusa-restore", true).Image == fmt.Sprintf("docker.io/%s:latest", medusaImageRepo), "dc1 doesn't have medusa-restore init container")
+
+	// Check volume mounts
```

Comment:
Consider using the `assert` variants for the all the volume mount and env var checks so that the test can check all of them and fail at the end.

Reply:
done 👍